x/incentive/keeper/abci.go

```diff
@@ -91,24 +99,90 @@ func (k Keeper) ProcessRewardsDistribution(ctx sdk.Context) {
 	}
 }
 
+func (k Keeper) ProcessUpdateIncentiveParams(ctx sdk.Context) bool {
+	listTimeBasedInflations := k.tokenomicsKeeper.GetAllTimeBasedInflation(ctx)
+	if len(listTimeBasedInflations) < 1 {
+		return false
+	}
+
+	params := k.GetParams(ctx)
+
+	for _, inflation := range listTimeBasedInflations {
+		if inflation.StartBlockHeight > uint64(ctx.BlockHeight()) || inflation.EndBlockHeight < uint64(ctx.BlockHeight()) {
+			continue
+		}
+
+		totalBlocksPerYear := sdk.NewInt(int64(inflation.EndBlockHeight - inflation.StartBlockHeight + 1))
+		allocationEpochInblocks := totalBlocksPerYear.Quo(sdk.NewInt(ptypes.DaysPerYear))
+		if len(params.LpIncentives) < 1 {
+			totalDistributionEpochPerYear := totalBlocksPerYear.Quo(sdk.NewInt(params.DistributionEpochForLpsInBlocks))
+			currentEpochInBlocks := sdk.NewInt(ctx.BlockHeight() - int64(inflation.StartBlockHeight)).Mul(totalDistributionEpochPerYear).Quo(totalBlocksPerYear)
+			maxEdenPerAllocation := sdk.NewInt(int64(inflation.Inflation.LmRewards)).Mul(allocationEpochInblocks).Quo(totalBlocksPerYear)
+			params.LpIncentives = append(params.LpIncentives, types.IncentiveInfo{
+				// reward amount in eden for 1 year
+				EdenAmountPerYear: sdk.NewInt(int64(inflation.Inflation.LmRewards)),
+				// starting block height of the distribution
+				DistributionStartBlock: sdk.NewInt(int64(inflation.StartBlockHeight)),
+				// distribution duration - block number per year
+				TotalBlocksPerYear: totalBlocksPerYear,
+				// we set block numbers in 24 hrs
+				AllocationEpochInBlocks: allocationEpochInblocks,
+				// maximum eden allocation per day that won't exceed 30% apr
+				MaxEdenPerAllocation: maxEdenPerAllocation,
+				// number of block intervals that distribute rewards.
+				DistributionEpochInBlocks: sdk.NewInt(params.DistributionEpochForLpsInBlocks),
+				// current epoch in block number
+				CurrentEpochInBlocks: currentEpochInBlocks,
+				// eden boost apr (0-1) range
+				EdenBoostApr: sdk.NewDec(1),
+			})
+		}
+
+		if len(params.StakeIncentives) < 1 {
```

Comment:
I think it would be good to update this to `== 0` for clarity

Reply:
Got it